cmd/usocksd/main.go

```diff
@@ -18,13 +18,24 @@ var (
 	optFile = flag.String("f", "", "configuration file name")
 )
 
+func serveMetrics(c *usocksd.Config) {
+	metricsServer := usocksd.NewMetricsServer(c)
+	mln, err := usocksd.MetricsListener(c)
+	if err != nil {
+		_ = log.Warn("could not initialize metrics server, but will proceed without it", map[string]interface{}{
+			"error": err.Error(),
+		})
+	}
+	metricsServer.Serve(mln)
```

Comment:
It should not try to serve metric if error is occurred.
If an error occurs when creating a listner, there is no guarantee that it will be able to listen, and it may panic.